pallets/remote-proxy/src/lib.rs

```diff
@@ -0,0 +1,354 @@
+// Copyright (C) Polkadot Fellows.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot. If not, see <http://www.gnu.org/licenses/>.
+
+extern crate alloc;
+
+#[cfg(test)]
+mod tests;
+
+use alloc::vec::Vec;
+use codec::{Encode, MaxEncodedLen};
+use frame_support::{storage::storage_prefix, Parameter, StorageHasher, Twox64Concat};
+use pallet_proxy::ProxyDefinition;
+use scale_info::TypeInfo;
+use sp_core::Hasher;
+use sp_runtime::traits::Saturating;
+
+pub use cumulus_primitives_core::PersistedValidationData;
+pub use pallet::*;
+
+/// The remote proxy interface.
+pub trait RemoteProxyInterface<AccountId, ProxyType, BlockNumber> {
+	/// The remote account id.
+	type RemoteAccountId: Parameter + MaxEncodedLen;
+	/// The remote proxy type.
+	type RemoteProxyType: Parameter + MaxEncodedLen;
+	/// The remote block number.
+	type RemoteBlockNumber: Parameter + Saturating + MaxEncodedLen + Default;
+	/// The hash type used by the remote chain.
+	type Hash: Parameter + MaxEncodedLen;
```

Comment:
this pallet is in the runtime repo instead of polkadot-sdk so presumably it is only intended to be used by runtimes here. in that case, do we really need this to be that generic?

Reply:
Not in the PSDK doesn't necessarily mean not reusable, thought.

Take for example [frame-contrib](https://github.com/virto-network/frame-contrib), a collection of pallets that are intended to be initially used in [Kreivo](https://github.com/virto-network/kreivo) but are generic and extensible enough so they can be used anywhere else.

Maybe by passing it in the future to the PSDK, or by being used by other chains, it's desirable to be generic.